_posts/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that.md

```diff
@@ -0,0 +1,265 @@
+---
+layout: post
+title: "Why should we rewrite enzyme.js to react-testing-library and how to do that?"
+author: [magdalena.mazur]
+tags: [tech, frontend, javascript, jest, enzyme, react-testing-library, rtl, react]
+---
+## Introduction
+Everyone repeats like a mantra that tests are an indispensable element of development work. Is there something about it? Well, I need to admit that as a developer, I rather often want to skip the test writing stage. I assume I'm not the only one. I am aware that it's a mistake, even with testers on board. Effective and efficient testing of your own code leads to catching bugs at the moment of creating a functionality, as well as when changing something already existing. It cannot be questioned. Sometimes tests also help to understand how some long-unused functionality or component works. And that's a small bonus too.
+**Can a large project cope without testing? Probably so. But the number of errors and their severity will probably be much higher.** That’s why in Allegro Ads we pay attention to writing tests.
+
+We have a fairly large number of frameworks that are used to write tests in JavaScript. We can list tools like: [MochaJS](https://mochajs.org/), [Jest](https://jestjs.io/), [Karma](https://karma-runner.github.io), [Jasmine](https://jasmine.github.io/), [Cypress](https://www.cypress.io/). Some of them focus more on business value, others on technical details. There are also a range of libraries which solve different problems. However, in this article, we will pay special attention and compare two testing javascript libraries like [Enzyme.js](https://enzymejs.github.io/enzyme/) and [React Testing Library](https://testing-library.com/docs/react-testing-library/intro/) (RTL).
```

Comment:
I would emphasize that React Testing Library is often called RTL because later on you are referencing it by that name and it is easy to miss just "(RTL)"